packages/server/graphql/mutations/helpers/bootstrapNewUser.ts

```diff
@@ -18,6 +18,28 @@ import sendPromptToJoinOrg from '../../../utils/sendPromptToJoinOrg'
 import {makeDefaultTeamName} from 'parabol-client/utils/makeDefaultTeamName'
 import isCompanyDomain from '../../../utils/isCompanyDomain'
 import {DataLoaderWorker} from '../../graphql'
+import getTeamsByIds from '../../../postgres/queries/getTeamsByIds'
+import acceptTeamInvitation from '../../../safeMutations/acceptTeamInvitation'
+import isValid from '../../isValid'
+import {Team} from '../../../postgres/queries/getTeamsByIds'
+
+const getAutoJoinTeams = async (
+  usersWithDomainTeams: Team[] | never[],
+  domain: string,
+  dataLoader: DataLoaderWorker
+) => {
+  const orgIds = usersWithDomainTeams
+    .filter(({autoJoin, isArchived}) => autoJoin && !isArchived)
```

Comment:
+1 I think we can filter this in database query using kysely. Otherwise, we always fetch all teams of all domain users every new sign up which feels unnecessarily heavy

Reply:
-1 I agree, we should do the `autoJoin` check on db side.